### DISCUSSION TITLE
final variable values and the functions values

### DISCUSSION THREAD

**Discussion by SamirLamin** [meta-pytorch/botorch#1690] [General]
I used betorch to solve my problem based on the existing problem integrated in the platform, can you please me show me how to get my final variable values and the functions values

**Comment by eytan** [comment]
Do you have a code example to reproduce? In our tutorials we plot the final
values at each iteration so it should be straightforward. You may wish to
try out Ax (http://ax.dev) if you are just using BoTorch as a tool for
optimization (for some real world application or research project). BoTorch
is intended for users who are developing new Bayesian optimization methods
(eg new acquisition functions).

Best
E

**Reply by SamirLamin** [reply]
I create my problem in multi_objective as a class like the BraninCurrin problem and I use it by replacing BraninCurrin by my problem with its characteristic I find the plots but I need to know how the final variable values and the values of the function optimum

**Reply by saitcakmak** [reply]
Hi @SamirLamin. Assuming you're following a tutorial like https://botorch.org/tutorials/multi_objective_bo, here's how you'd get the points on the Pareto frontier. From that tutorial, suppose that you picked qNEHVI and have access to `train_x_qnehvi` and `train_obj_qnehvi` after the BO loop completed. You can use `is_non_dominated` to find which points are on the Pareto frontier:
```
from botorch.utils.multi_objective.pareto import is_non_dominated
non_dominated_flag = is_non_dominated(train_obj_qnehvi)  # boolean denoting whether the corresponding objective is non-dominated.
pareto_x = train_x_qnehvi[non_dominated_flag]
pareto_obj = train_obj_qnehvi[non_dominated_flag]
```
`pareto_x` gives you the designs and the `pareto_obj` the corresponding objective values of the designs on the observed Pareto frontier. Note that since this is a multi-objective problem, there isn't a single optimal value but a set of points that are on the observed Pareto frontier.

**Reply by SamirLamin** [reply]
Ok Thank you so much for your help
